deliver/lib/deliver/app_screenshot_iterator.rb

```diff
@@ -0,0 +1,98 @@
+module Deliver
+  # This is a convinient class that enumerates app store connect's screenshots in various degrees.
+  class AppScreenshotIterator
+    NUMBER_OF_THREADS = Helper.test? ? 1 : 10
+
+    # @param localizations [Array<Spaceship::ConnectAPI::AppStoreVersionLocalization>]
+    def initialize(localizations)
+      @localizations = localizations
+    end
+
+    # Iterate app_screenshot_set over localizations
+    #
+    # @yield [localization, app_screenshot_set]
+    # @yieldparam [optional, Spaceship::ConnectAPI::AppStoreVersionLocalization] localization
+    # @yieldparam [optional, Spaceship::ConnectAPI::AppStoreScreenshotSet] app_screenshot_set
+    def each_app_screenshot_set(&block)
+      return enum_for(__method__) unless block_given?
+
+      # Collect app_screenshot_sets from localizations in parallel but
+      # limit the number of threads working at a time with using `lazy` and `force` controls
+      # to not attack App Store Connect
+      results = @localizations.each_slice(NUMBER_OF_THREADS).lazy.map do |localizations|
+        localizations.map do |localization|
+          Thread.new do
+            [localization, localization.get_app_screenshot_sets]
+          end
+        end
+      end.flat_map do |threads|
+        threads.map { |t| t.join.value }
+      end.force
```

Comment:
FYI: This may sound too tricky but you can confirm the behaviour with this snippet. I was a bit afraid that a project that supports many localisations may cause DDoS-ish attack here. What this cares here is essentially same as what `QueueWorker` cares but it wasn't suitable for collecting value so that I quickly hacked this. 


```ruby
a = [*1..10].each_slice(3).lazy.map do |slice|
  slice.map do |x|
    Thread.new do
      puts "start #{x}"
      sleep(3)
      puts "end #{x}"
      x
    end
  end
end.flat_map do |threads|
  threads.map { |t| t.join.value }
end.force

p a
```

Reply:
This seems good to me! I don't think I've seen any kind of rate limit yet so I think we should be good. But maybe we could introduce an optional environment variable for overriding if somebody wants to do less? (see suggestion above)

Reply:
> we could introduce an optional environment variable for overriding if somebody wants to do less

Sounds good. I'll add it.